scala-package/core/src/main/scala/org/apache/mxnet/io/NDArrayIter.scala

```diff
@@ -42,27 +43,40 @@ import scala.collection.immutable.ListMap
 class NDArrayIter(data: IndexedSeq[(String, NDArray)],
                   label: IndexedSeq[(String, NDArray)],
                   private val dataBatchSize: Int, shuffle: Boolean,
-                  lastBatchHandle: String) extends DataIter {
-
+                  lastBatchHandle: String,
+                  dataDType: DType, labelDType: DType,
```

Comment:
With this change all inputs/labels will have the same DType and Label, however data and labels can take IndexedSeq[(String, NDArray)] -- this does not seem correct. 

Can the NDArrayIter constructor be
```
class NDArrayIter(data: IndexedSeq[NDArray], 
label: IndexedSeq[NDArray], 
val dataDesc: IndexedSeq[DataDesc],
val labelDesc: IndexedSeq[DataDesc],
val dataBatchSize: Int,
lastBatchHandle: String)
```

Reply:
Can we put the DType and Label in the same IndexedSeq as the NDArray? It would help make this a little less prone to bugs because they two would be coupled together.

Something like:
```scala
class NDArrayIter(data: IndexedSeq[(DataDesc, NDArray)],
label: IndexSeq[(DataDesc, NDArray)],
val dataBatchSize: Int,
lastBatchHandle: String)
```

Reply:
like this better.